### DISCUSSION TITLE
What's the idiomatic way of evaluating a rule once, but without any class scanning?

### DISCUSSION THREAD

**Discussion by pivovarit** [TNG/ArchUnit#1391] [Q&A] [answered]
We have a rule that we'd like to enforce project-wide - that would be a consistency of some files that should be present in multiple directories. 

We could trick the analyzer to do `ArchRuleDefinition.theClass(Object.class)` or something equally hacky, but is there an idiomatic solution that doesn't involve scanning classes which can be lengthy?

**Comment by hankem** [comment]
The rule definition is independent of the class file import. The latter is described in [§6.1 Import](https://www.archunit.org/userguide/html/000_Index.html#_import) of the ArchUnit User Guide.

If I understood correctly, you want to optimize the import to skip lengthy classes and only include _some_ files? This should be easily doable with a custom `ImportOption`.

**Reply by pivovarit** [reply]
I'm aware of that, but the thing is... I don't really want to analyze any Java classes. It's just a generic standalone rule, which would be more/less: `ensure that matching migration files are present in all directories`. Those migration files are plain SQL files.

Technically, I don't need ArchUnit to implement that, but it would be convenient to have the ability to have such a check included as part of the arch-unit suite.

**Reply by hankem** [reply, marked as answer]
Hm, I see. Then ArchUnit might currently not be the best fit for your problem. I'm not even sure whether you could express a test that SQL files must be present, as `ArchRule`s can only operate on (information derived from) imported `JavaClasses`.

**Reply by pivovarit** [reply]
Ok, thanks for the clarification - my main motivation for using ArchUnit was the ability to ship this rule alongside N other standard rules
